test_capability_validation.py

```
@@ -0,0 +1,60 @@
+    limited_runtime = RuntimeCapabilityMatrix(
+        basic_chat=True,
+        simple_tools=True,
+        # Missing: native_hooks, streaming_deltas, etc.
+    )
+    
+    # Require capabilities the limited runtime doesn't have
+    required = {RuntimeCapability.NATIVE_HOOKS, RuntimeCapability.STREAMING_DELTAS}
+    
+    try:
+        validate_capabilities(limited_runtime, required, "plugin-harness")
+        assert False, "Expected CapabilityValidationError but validation passed"
```

Ruff (0.15.17)

[warning] Do not `assert False` (`python -O` removes these calls), raise `AssertionError()`

Replace `assert False`

(B011)